plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlScanner.java

```diff
@@ -64,9 +80,44 @@ public void characters(char ch[], int start, int length) throws SAXException {
   public void endElement(String uri, String localName, String qName) throws SAXException {
     if ("servlet-class".equalsIgnoreCase(localName)) {
       insideServletClass = false;
-      String servletClass = servletClassContents.toString();
-      addToBlacklist(new UndefinedServletElement(
-          servletClass, servletClassLocation, localName.length() + 2));
+      String servletClassName = servletClassContents.toString();
+      if (findClass(servletClassName) == null) {
+        addToBlacklist(new UndefinedServletElement(
+            servletClassName, servletClassLocation, localName.length() + 2));
+      }
+    }
+  }
+  
+  /**
+   * Checks for the given class name within the project.
+   */
+  static IType findClass(String typeName) {
+    if (typeName == null || "".equals(typeName)) {
+      return null;
+    }
+    SearchPattern pattern = SearchPattern.createPattern(typeName, 
+        IJavaSearchConstants.CLASS,
+        IJavaSearchConstants.DECLARATIONS,
+        SearchPattern.R_EXACT_MATCH | SearchPattern.R_ERASURE_MATCH);
+    IJavaSearchScope scope = SearchEngine.createWorkspaceScope();
+    CollectingSearchRequestor requestor = new CollectingSearchRequestor();
+    performSearch(pattern, scope, requestor, null);
+    List<SearchMatch> results = requestor.getResults();
+    return results.isEmpty() ? null : (IType) results.get(0).getElement();
+  }
+  
+  /**
+   * Performs the search for a class of a given pattern within the project.
```

Comment:
performs the search for a class of a given pattern --> Searches for a class that matches a pattern